include/open62541/server_pubsub.h

```diff
@@ -492,6 +492,48 @@ UA_Server_getDataSetWriterConfig(UA_Server *server, const UA_NodeId dsw,
 UA_StatusCode UA_EXPORT
 UA_Server_removeDataSetWriter(UA_Server *server, const UA_NodeId dsw);
 
+/**
+ * SubscribedDataSet
+ * -----------------
+ * SubscribedDataSet describes the processing of the received DataSet. SubscribedDataSet defines which field
+ * in the DataSet is mapped to which Variable in the OPC UA Application. SubscribedDataSet has two sub-types
+ * called the TargetVariablesType and SubscribedDataSetMirrorType.
+ * SubscribedDataSetMirrorType is currently not supported. SubscribedDataSet is set to TargetVariablesType
+ * and then the list of target Variables are created in the Subscriber AddressSpace.
+ * TargetVariables are a list of variables that are to be added in the Subscriber AddressSpace.
+ * It defines a list of Variable mappings between received DataSet fields and added Variables
+ * in the Subscriber AddressSpace. */
+
+typedef struct {
+    UA_Guid dataSetFieldId;
+    UA_String receiverIndexRange;
+    UA_NodeId targetNodeId;
+    UA_UInt32 attributeId;
+    UA_String writeIndexRange;
+    UA_OverrideValueHandling overrideValueHandling;
+    UA_Variant overrideValue;
+    // reference to the target node value backend pointer if rt fixed size is enabled
+    UA_DataValue ** externalDataValueSource;
+} UA_FieldTargetVariablesSourceEntry;
+
+typedef struct {
+    size_t targetVariablesSize;
+    UA_FieldTargetVariablesSourceEntry *targetVariables; // Can be modifed to use LIST or TAILQ
+} UA_TargetVariablesSourceEntry;
+
+void UA_EXPORT
+UA_TargetVariablesSource_clear(UA_TargetVariablesSourceEntry *subscribedDataSetTarget);
+
+/* Return Status Code after creating TargetVariables in Subscriber AddressSpace */
+UA_StatusCode UA_EXPORT
+UA_Server_DataSetReader_createTargetVariables(UA_Server *server, UA_NodeId dataSetReaderIdentifier,
+                                              UA_TargetVariablesSourceEntry* targetVariables);
```

Comment:
Personally, I would prefer to use the UA_TargetVariablesSource * and a size as input parameter in our API. The **'UA_TargetVariablesSourceEntry** is more internally.